src/main/scala/intellij/haskell/external/repl/ProjectStackReplProcess.scala

```diff
@@ -47,20 +48,24 @@ private[repl] class ProjectStackReplProcess(project: Project) extends StackReplP
   }
 
   private def load(psiFile: PsiFile, moduleName: Option[String]): Option[(StackReplOutput, Boolean)] = {
-    val filePath = getFilePath(psiFile)
-    val moduleNameAction = ApplicationManager.getApplication.runReadAction(new Computable[Option[String]] {
-      override def compute(): Option[String] = {
-        moduleName.orElse(HaskellPsiUtil.findModuleName(psiFile))
+    if (!HaskellConsoleView.isConsole(psiFile)) {
```

Comment:
Okay, but this is wrong place to check. Take a look to HaskellAnnotator.

Also, because it's a psi file, are you sure no other calls are made to this project  repl?